src/main/java/com/ea/restaurant/grpc/GrpcOauth2ServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.ea.restaurant.grpc;
+
+import com.ea.restaurant.grpc.interceptors.GrpcServerInterceptor;
+import com.ea.restaurant.service.Oauth2Service;
+import com.ea.restaurant.util.GrpcLoginClientResponseMapper;
+import com.ea.restaurant.util.GrpcUtil;
+import io.grpc.Metadata;
+import io.grpc.stub.StreamObserver;
+import java.util.Objects;
+import net.devh.boot.grpc.server.service.GrpcService;
+
+@GrpcService
+public class GrpcOauth2ServiceImpl extends Oauth2ServiceGrpc.Oauth2ServiceImplBase {
+  private final Oauth2Service oauth2Service;
+
+  public GrpcOauth2ServiceImpl(Oauth2Service oauth2Service) {
+    this.oauth2Service = oauth2Service;
+  }
+
+  @Override
+  public void loginClient(
+      LoginClientRequest request, StreamObserver<LoginClientResponse> responseObserver) {
+
+    var authorizationValue =
+        ((Metadata) GrpcServerInterceptor.CONTEXT_METADATA.get())
+            .get(Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER));
```

Comment:
Use enums or constant static strings in order to get the metadata key name

Reply:
Done.